template.yaml

```diff
@@ -323,70 +324,93 @@ Resources:
-              Action:
-                - sqs:SendMessage
-              Resource:
-                - !GetAtt SendEmailQueue.Arn
-            - Sid: UseSqsKmsKey
-              Effect: Allow
-              Action:
-                - kms:Decrypt
-                - kms:GenerateDataKey*
-                - kms:ReEncrypt*
-              Resource: '{{resolve:ssm:SqsKmsKeyArn}}'
+
   GenerateDownloadFunctionLogs:
     Type: AWS::Logs::LogGroup
     Properties:
       KmsKeyId: '{{resolve:ssm:LogsKmsKeyArn}}'
       LogGroupName: !Sub '/aws/lambda/${AWS::StackName}-generate-download'
       RetentionInDays: 30
+
+  GenerateDownloadFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub ${AWS::StackName}-${Environment}-generate-download-role
+      PermissionsBoundary:
+        !If [
+          UsePermissionsBoundary,
+          !Ref PermissionsBoundary,
+          !Ref AWS::NoValue
+        ]
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Policies:
+        - PolicyName: AllowGenerateDownloadActions
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Sid: SecureFraudSiteDataTableWrite
+                Effect: Allow
+                Action:
+                  - dynamodb:PutItem
+                Resource: '{{resolve:ssm:SecureFraudSiteDataTableArn}}'
+              - Sid: DecryptDatabaseKmsKey
+                Effect: Allow
+                Action:
+                  - kms:Decrypt
+                Resource: '{{resolve:ssm:DatabaseKmsKeyArn}}'
+              - Sid: Logs
+                Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: !GetAtt GenerateDownloadFunctionLogs.Arn
+              - Sid: AllowQueueRead
+                Effect: Allow
+                Action:
+                  - sqs:ReceiveMessage
+                  - sqs:DeleteMessage
+                  - sqs:GetQueueAttributes
+                Resource: !Sub 'arn:aws:sqs:${AWS::Region}:{{resolve:ssm:AuditAccountNumber}}:txma-data-analysis-${Environment}-query-completed-queue'
+              - Sid: UseQueryCompletedQueueKmsKey
+                Effect: Allow
+                Action:
+                  - kms:Decrypt
+                  - kms:GenerateDataKey*
+                  - kms:ReEncrypt*
+                Resource: !Sub 'arn:aws:kms:${AWS::Region}:{{resolve:ssm:AuditAccountNumber}}:*'
+              - Sid: AllowAccessAthenaOutputBucket
+                Effect: Allow
+                Action:
+                  - s3:GetObject
+                Resource: !Sub 'arn:aws:s3:::txma-data-analysis-${Environment}-athena-query-output-bucket/*'
+              - Sid: AllowDecryptOfAthenaOutput
```

Comment:
You can delete this as it is a duplicate of `UseQueryCompletedQueueKmsKey`. I.e. the `kms:Decrypt` permission is already applied to this resource pattern.